spec/requests/topics_controller_spec.rb

```diff
@@ -2519,6 +2519,12 @@ def extract_post_stream
       expect(response.body).to_not include("/forum/forum")
       expect(response.body).to include("http://test.localhost/forum/t/#{topic.slug}")
     end
+
+    it 'returns 404 when posts are deleted' do
+      topic.posts.map(&:trash!)
```

Comment:
You don't need to `map` when you don't use the return value 😉 

```suggestion
      topic.posts.each(&:trash!)
```